ocs_ci/utility/rosa.py

```diff
@@ -95,14 +97,12 @@ def get_providers_subnet(region, cluster_name):
 
 
     """
-    subnets = config.ENV_DATA.get("subnet_id", "")
-    # TODO:  This is Temporary function Fix to get subnet ids from config and
-    # its actual implementation shall be done using aws API
-    # In manual method we use below command to get the subnet id
-    # subnet_cmd = (f"/usr/bin/aws --region {region} ec2 describe-subnets"
-    #               f" --filters Name=tag:Name,Values={cluster_name}*"
-    #               f" --output text | grep subnet |awk '{print $15}'")
-    return subnets
+    ec2_client = boto3.client("ec2", region_name=region)
+    subnets = ec2_client.describe_subnets(
+        Filters=[{"Name": "tag:Name", "Values": [f"{cluster_name}*"]}]
+    )
```

Comment:
It will be little bit more work, but I would propose to put this functionality to [`ocs_ci/utility/aws.py`](https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/utility/aws.py) to have all AWS related methods/functions together.